### DISCUSSION TITLE
_jsxDevRuntime.jsxDEV is not function

### DISCUSSION THREAD

**Discussion by monic-shixi** [babel/babel#13013] [General] [answered]
I had update webpack4 to webpack, but `@babel/preset-react` not work in dev(webpack4 dev work).

- env
  - @babel/preset-react@7.12.7
  - react@16.14.0

```js
presets: [
  [
    '@babel/preset-react',
    {
      runtime: 'automatic',
      development: process.env.BABEL_ENV !== 'production'
    }
  ],
]
```
For example:
```js
// example.js
export default class Example extends React.Component {
  render() {
    return <div />
  }
}
// a.js
const example = <Example />
```
It's throw a error: 
```js
Uncaught TypeError: (0 , _jsxDevRuntime.jsxDEV) is not a function
```
 
Or this is a bug?

**Comment by JLHwung** [comment, marked as answer]
Does it work with `runtime: "classic"`? The JSX dev runtime is supported in 16.14 and above, did you check if you have multiple react in node_modules?

**Reply by monic-shixi** [reply]
Thanks for your response.
`runtime: 'classic'` work. And `require.resolve('react/jsx-dev-runtime')` it's ok.
`yarn list --depth=0 | grep react@` only one 16.14.0 version.
So, it's may be a webpack5 bug?

**Reply by monic-shixi** [reply]
I'm re-check react verison without `--depth=0` argument. see another version react@16.8.6. 
And i'm add resolve.alias in webpack config:
```js
resolve: {
  alias: {
    react: path.resolve(__dirname, './node_modules/react')
  }
}
```
then it seems work.

Thank you very muck.

**Reply by JLHwung** [reply]
Glad that helps!

**Comment by ramseycoder** [comment]
I use the previous responses but i have always the error ..
i use vite 2.8.0 and react 17.0.2

**Comment by KKrisu** [comment]
on the other hand I can see this error when using the swc compiler :thinking:

**Reply by nicolo-ribaudo** [reply]
If you have the https://swc.rs/docs/configuration/compilation#jsctransformreact option set to `automatic`, chance it to `classic`. If it's already set to classic, one of your dependencies probably requires a newer version of react than the one you are using (React 17 should work).

Otherwise you could try asking in the swc repository, since this one is about Babel 😅
